File: 프로그래머스/소수 만들기.py
from itertools import combinations
# 소수 판별 함수
def primeNumber(n):
    if n == 0 or n == 1:
        return False
    
    for i in range(2, (n // 2) + 1):
        if n % i == 0:
            return False
        
    return True
    
def solution(nums):
    # 3개 숫자 조합 모든 경우의 수 튜플 리스트로
    case = list(combinations(nums, 3))
    result = 0
    for t in case:
        if primeNumber(sum(t)):
            result += 1
    
    return result

File: 프로그래머스/소수 만들기.java
class Solution {
    public boolean isPrime(int num) {
        if (num == 1)
            return false;
        for (int i = 2; i <= num/2; i++) {
            if (num % i == 0)
                return false;
        }
        return true;
    }

    public int solution(int[] nums) {
        int answer = 0;
        for (int i = 0; i < nums.length-2; i++) {
            for (int j = i+1; j < nums.length-1; j++) {
                for (int k = j+1; k < nums.length; k++) {
                    if (isPrime(nums[i] + nums[j] + nums[k]))
                        answer++;
                }
            }
        }
        return answer;
    }
}